src/katgpucbf/xbgpu/engine.py

```diff
@@ -455,15 +454,29 @@ def __init__(
             tx_enabled=self._init_tx_enabled,
         )
 
-    def populate_sensors(self, sensors: aiokatcp.SensorSet, rx_sensor_timeout: float) -> None:
-        """Define the sensors for an XBEngine."""
+    def populate_sensors(
+        self,
+        sensors: aiokatcp.SensorSet,
+        rx_sensor_timeout: float,
+        chan_range: tuple[int, int],
+    ) -> None:
+        """Define the sensors for an XBEngine.
+
+        Parameters
+        ----------
+        rx_sensor_timeout
+            See :meth:`.recv.make_sensors` for more information.
+        chan_range
+            Tuple of integers showing the two values of the chan-range sensor
+            - (channel_offset, channel_offset + channels_per_substream)
+        """
         # Static sensors
         sensors.add(
             aiokatcp.Sensor(
                 str,
                 "chan-range",
                 "The range of channels processed by this XB-engine, inclusive",
-                default=f"({self.channel_offset_value},{self.channel_offset_value + self.n_channels_per_stream - 1})",
+                default=f"({chan_range[0]},{chan_range[1] - 1})",
```

Comment:
I don't really see the reason for this change?

Reply:
In my original move to make `channels_per_substream` part of an `{X, B}Output`
* I realised that the attribute would no longer be part of the `XBEngine`, and 
* therefore wouldn't be accessible as `self.<something>` in `populate_sensors`.

I figured making `populate_sensors` accept the channel range as an arg would be a straightforward solution.
* Even setting aside the `{X, B}Output` conundrum, is it ~good practice to make an arg an attribute even if it's only used once elsewhere outside the scope of `__init__`?

Reply:
Ok, I see the logic now.

It would break from the convention of the other `populate_sensors` functions though.

Reply:
Yeah that is the concern, maybe I'll roll it back for now and earmark it for addition/update during NGC-929.